### PR TITLE
feat: add monitoring-thanos-logs plugin

### PR DESCRIPTION
**Pull request by cy00r**
# Add Monitoring Thanos Logs Plugin Support

## Overview

This PR adds support for installing and managing the **monitoring-thanos-logs** plugin, which provides centralized log aggregation and querying capabilities using Loki, Alloy, and Grafana integration.

### Features Added

1. **Thanos Logs Stack Installation**
   - Automatic installation of Loki (log aggregation system) via Helm
   - Automatic installation of Alloy (log collection agent) via Helm
   - S3 bucket creation for log storage (chunks, ruler, admin buckets)
   - Automatic Grafana datasource configuration for Loki
   - Dependency check: Ensures Grafana is installed before proceeding

2. **Thanos Logs Stack Uninstallation**
   - Clean removal of Helm releases (Loki and Alloy)
   - Optional S3 bucket deletion with user confirmation
   - Automatic removal of Loki datasource from Grafana
   - Integration with chain destroy process

3. **Code Improvements**
   - Refactored Grafana password input into reusable `InputGrafanaPassword()` function
   - Added new types: `ThanosLogsConfig` and `ThanosLogsHelmValues`
   - Added namespace constant: `MonitoringThanosLogsNamespace`
   - Added plugin constant: `PluginMonitoringThanosLogs`

### Prerequisites
1. A deployed chain with K8s configuration
2. AWS credentials configured with S3 access
3. Helm installed and configured
4. kubectl access to the cluster

### Installation Test

1. **Install the monitoring-thanos-logs plugin:**
   ```bash
   trh-sdk install monitoring-thanos-logs
   ```

2. **During installation, you will be prompted for:**
   - Grafana admin password (required)

3. **Verify installation:**
   ```bash
   # Check Helm releases
   helm list -n monitoring
   
   # Should show releases like:
   # - thanos-logs-<timestamp>
   # - alloy-<timestamp>
   
   # Check pods
   kubectl get pods -n monitoring
   
   # Should show pods like:
   # - <release-name>-gateway-*
   # - <release-name>-read-*
   # - <release-name>-write-*
   # - alloy-*
   ```

4. **Verify S3 buckets created:**
   ```bash
   aws s3 ls | grep thanos-logs
   
   # Should show:
   # <chain-name>-thanos-logs-chunks
   # <chain-name>-thanos-logs-ruler
   # <chain-name>-thanos-logs-admin
   ```

5. **Verify Grafana datasource:**
   - Access Grafana UI (get URL from `trh-sdk info`)
   - Navigate to Configuration > Data Sources
   - Verify "Loki" datasource is present and configured

6. **Test log querying in Grafana:**
   - Go to Explore in Grafana
   - Select Loki datasource
   - Try a simple query like: `{namespace="monitoring"}`

### Uninstallation Test

1. **Uninstall the plugin:**
   ```bash
   trh-sdk uninstall monitoring-thanos-logs
   ```

2. **When prompted, choose whether to delete S3 buckets:**
   - Type `y` to delete buckets (for cleanup)
   - Type `n` to keep buckets (for data preservation)

3. **Verify uninstallation:**
   ```bash
   # Check Helm releases (should be empty)
   helm list -n monitoring | grep -E "thanos-logs|alloy"
   
   # Check pods (should be gone)
   kubectl get pods -n monitoring | grep -E "loki|alloy"
   ```

4. **Verify Grafana datasource removed:**
   - Access Grafana UI
   - Navigate to Configuration > Data Sources
   - Verify "Loki" datasource is no longer present

### Integration Test

1. **Test with destroy command:**
   ```bash
   # Install the plugin first
   trh-sdk install monitoring-thanos-logs
   
   # Destroy the chain (should also clean up thanos-logs)
   trh-sdk destroy
   
   # Verify cleanup
   helm list -n monitoring | grep -E "thanos-logs|alloy"
   ```

### Edge Cases to Test

1. **Install when Grafana is not installed:**
   - The plugin should automatically install Grafana with default monitoring config
   - Then proceed with Thanos logs installation

2. **Install when already installed:**
   - Running install again should detect existing installation
   - Should skip installation gracefully

3. **Install with invalid Grafana password:**
   - Should handle errors gracefully
   - Should not leave partial installations

4. **Uninstall when not installed:**
   - Should handle gracefully without errors

## Configuration

The plugin uses the following configuration:
- **Namespace**: `monitoring` (shared with monitoring plugin)
- **S3 Buckets**: 
  - `<chain-name>-thanos-logs-chunks`
  - `<chain-name>-thanos-logs-ruler`
  - `<chain-name>-thanos-logs-admin`
- **Helm Release Names**: Auto-generated with timestamps

## Dependencies

- Requires Grafana to be installed (will auto-install if not present)
- Requires AWS S3 access for log storage
- Requires Helm charts in `tokamak-thanos-stack/charts/loki` and `tokamak-thanos-stack/charts/alloy`

## Notes

- The plugin shares the `monitoring` namespace with the monitoring plugin
- S3 buckets are created per chain name

### CODE REVIEW

File: pkg/stacks/thanos/monitoring_thanos_logs.go [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+	}
+
+	if strings.Contains(output, "error") || strings.Contains(output, "Error") || strings.Contains(output, "HTTP_CODE:4") || strings.Contains(output, "HTTP_CODE:5") {
+		logger.Warnw("Error response from Grafana API", "output", output)
+		return fmt.Errorf("Grafana API returned an error: %s", output)
+	}
+
+	logger.Info("Successfully added Loki datasource to Grafana")
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (cy00r) [reply pkg/stacks/thanos/monitoring_thanos_logs.go]: Fixed in latest commit.

File: pkg/stacks/thanos/monitoring_thanos_logs.go [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+	// Check if deletion was successful (200 OK or 404 Not Found)
+	if strings.Contains(output, "HTTP_CODE:200") || strings.Contains(output, "HTTP_CODE:404") {
+		logger.Info("Loki datasource removed from Grafana")
+		return nil
+	}
+
+	// If deletion failed, log warning but don't fail the uninstall
+	logger.Warnw("Failed to remove Loki datasource from Grafana", "output", output)
+	return nil
+}
```
[the remark this answers is not in the document]
Comment (cy00r) [reply pkg/stacks/thanos/monitoring_thanos_logs.go]: Fixed in latest commit.

File: pkg/stacks/thanos/monitoring_thanos_logs.go [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
+	if !grafanaInstalled {
+		logger.Info("Grafana is not installed. Installing monitoring plugin...")
+		if err := t.installMonitoringWithDefaults(ctx, config.GrafanaPassword); err != nil {
+			logger.Errorw("Failed to install monitoring plugin", "err", err)
+			return "", fmt.Errorf("failed to install monitoring plugin: %w", err)
+		}
+
+		// Wait a bit for Grafana to be ready
+		logger.Info("Waiting for Grafana to be ready...")
+		time.Sleep(10 * time.Second)
```
Comment (0x6e616d) [inline pkg/stacks/thanos/monitoring_thanos_logs.go]: Could we check with the status or the health api?

File: pkg/stacks/thanos/monitoring_thanos_logs.go [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
+	if err != nil {
+		logger.Errorw("Failed to install Loki", "err", err, "output", out)
+		return "", fmt.Errorf("failed to install Loki: %w", err)
+	}
+
+	logger.Info("✅ Loki installed successfully")
+
+	// Wait for Loki to be ready
+	logger.Info("Waiting for Loki to be ready...")
+	time.Sleep(10 * time.Second)
```
Comment (0x6e616d) [inline pkg/stacks/thanos/monitoring_thanos_logs.go]: same above, could we check the k8s resources or api health

File: pkg/stacks/thanos/monitoring_thanos_logs.go [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+func (t *ThanosStack) configureLokiDatasourceInGrafana(ctx context.Context, config *types.ThanosLogsConfig) error {
+	logger := t.getLogger()
+
+	// Get Grafana pod name using GetPodNamesByLabel
+	grafanaPods, err := utils.GetPodNamesByLabel(ctx, constants.MonitoringNamespace, "app.kubernetes.io/name=grafana")
+	if err != nil || len(grafanaPods) == 0 {
+		return fmt.Errorf("failed to find Grafana pod: %w", err)
+	}
+
+	grafanaPodName := grafanaPods[0]
```
Comment (0x6e616d) [inline pkg/stacks/thanos/monitoring_thanos_logs.go]: check the length
